$OpenBSD: patch-tools_clang_lib_Driver_ToolChains_h,v 1.10 2014/07/16 22:16:54 brad Exp $

r209432
Use stack protector strong by default on OpenBSD

r211075
Use the integrated assembler by default on OpenBSD/powerpc

r213105
Use the integrated assembler by default on OpenBSD/sparc

--- tools/clang/lib/Driver/ToolChains.h.orig	Sun Mar  2 22:03:41 2014
+++ tools/clang/lib/Driver/ToolChains.h	Fri Jul 11 22:45:01 2014
@@ -538,7 +538,15 @@ class LLVM_LIBRARY_VISIBILITY OpenBSD : public Generic
   virtual bool isPIEDefault() const { return true; }
 
   virtual unsigned GetDefaultStackProtectorLevel(bool KernelOrKext) const {
-    return 1;
+    return 2;
+  }
+
+  virtual bool IsIntegratedAssemblerDefault() const {
+    if (getTriple().getArch() == llvm::Triple::ppc ||
+        getTriple().getArch() == llvm::Triple::sparc ||
+        getTriple().getArch() == llvm::Triple::sparcv9)
+      return true;
+    return Generic_ELF::IsIntegratedAssemblerDefault();
   }
 
 protected:
